Sources/Valkey/Connection/ValkeyConnection.swift

```diff
@@ -328,15 +328,13 @@ public final actor ValkeyConnection: ValkeyConnectionProtocol, Sendable {
     /// create a BSD sockets based bootstrap
     private static func createSocketsBootstrap(eventLoopGroup: EventLoopGroup) -> ClientBootstrap {
         ClientBootstrap(group: eventLoopGroup)
-            .channelOption(ChannelOptions.allowRemoteHalfClosure, value: true)
```

Comment:
The client doesn't need half closure, and it meant we were leaving connections open if the database was closed